manticore/core/smtlib/solver.py

```diff
@@ -396,6 +397,52 @@ def _pop(self):
         """Recall the last pushed constraint store and state."""
         self._smtlib.send("(pop 1)")
 
+    @lru_cache(maxsize=32)
+    def get_model(self, constraints: ConstraintSet):
+        self._reset(constraints.to_string())
+        self._smtlib.send("(check-sat)")
+        self._smtlib.recv()
+
+        model = {}
+        for variable in constraints.variables:
+            value = None
+            if isinstance(variable, BoolVariable):
+                value = self.__getvalue_bool(variable.name)
+            elif isinstance(variable, BitvecVariable):
+                value = self.__getvalue_bv(variable.name)
+            elif isinstance(variable, Array):
+                try:
+                    if variable.length is not None:
+                        value = []
+                        for i in range(len(variable)):
+                            variable_i = variable[i]
+                            if issymbolic(variable_i):
+                                value.append(self.__getvalue_bv(translate_to_smtlib(variable_i)))
+                            else:
+                                value.append(variable_i)
+                        value = bytes(value)
+                    else:
+                        # Only works if we know the max index of the arrray
+                        used_indexes = map(self.__getvalue_bv, variable.written)
+                        valued = {}
+                        for i in used_indexes:
+                            valued[i] = self.__getvalue_bv(variable[i])
+
+                        class A:
+                            def __init__(self, d, default):
+                                self._d = d
+                                self._default = default
+
+                            def __getitem__(self, index):
+                                return self._d.get(index, self._default)
+
+                        value = A(valued, variable.default)
```

Comment:
Why is this class here? Can you add a comment?

Reply:
`get_model` needs doc if it going to stay.

The goal of this method is to get an assigment for all variables defined in the CS.
There is a `problem` with Array variables. Because the SMT solver finds assigments for the "used" slots and assume a value for all the unused slots or something like that. Need to paste an example of what smtlib produces as a assigment for an array variable but I think the result constant array assignment is:

* unbounded
* have a default value for unused slots

So, not that the A class is close to a good solution but we can not simply return a "bytes" instance here.